app/controllers/events_controller.rb

```diff
@@ -62,4 +63,15 @@ def param_converter_options
     { skip: [:data_file_ids] }
   end
 
+  def set_timezone
+    time_zone = params[:event][:time_zone]
+    if time_zone.present? && (ActiveSupport::TimeZone.all.map{ |t| t.tzinfo.name }.include? time_zone)
+      start_date = params[:event][:start_date]
+      end_date = params[:event][:end_date]
+
+      params[:event][:start_date] = start_date.in_time_zone(time_zone) if start_date.present?
+      params[:event][:end_date] = end_date.in_time_zone(time_zone) if end_date.present?
```

Comment:
Won't this mean the start/end times will shift each time the event is updated?

Since it will re-apply the timezone modification